huaweicloud/resource_huaweicloud_cce_addon_v3.go

```diff
@@ -0,0 +1,205 @@
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func resourceCCEAddonV3() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceCCEAddonV3Create,
+		Read:   resourceCCEAddonV3Read,
+		Delete: resourceCCEAddonV3Delete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(10 * time.Minute),
+			Delete: schema.DefaultTimeout(3 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{ // request and response parameters
+			"authenticating_proxy_ca": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"cluster_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"addon_version": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"addon_template_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"region": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceCCEAddonV3Create(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	cceClient, err := config.cceAddonV3Client(GetRegion(d, config))
+
+	if err != nil {
+		return fmt.Errorf("Unable to create HuaweiCloud CCE client : %s", err)
+	}
+	authenticating_proxy := make(map[string]string)
+	if hasFilledOpt(d, "authenticating_proxy_ca") {
+		authenticating_proxy["ca"] = d.Get("authenticating_proxy_ca").(string)
+	}
+	createOpts := addons.CreateOpts{
+		Kind:       "Addon",
+		ApiVersion: "v3",
+		Metadata: addons.CreateMetadata{
+			Anno: addons.Annotations{
+				AddonInstallType: "install",
+			},
+		},
+		Spec: addons.RequestSpec{
+			Version:           d.Get("addon_version").(string),
+			ClusterID:         d.Get("cluster_id").(string),
+			AddonTemplateName: d.Get("addon_template_name").(string),
+			Values: addons.Values{
+				Basic: map[string]string{},
+			},
+		},
+	}
+
+	create, err := addons.Create(cceClient, createOpts, d.Get("cluster_id").(string)).Extract()
+
+	if err != nil {
+		return fmt.Errorf("Error creating HuaweiCloud CCEAddon: %s", err)
+	}
+
+	log.Printf("[DEBUG] Waiting for HuaweiCloud CCEAddon (%s) to become available", create.Metadata.Id)
+
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"installing", "abnormal"},
+		Target:     []string{"running"},
+		Refresh:    waitForCCEAddonActive(cceClient, create.Metadata.Id, d.Get("cluster_id").(string)),
+		Timeout:    d.Timeout(schema.TimeoutCreate),
+		Delay:      5 * time.Second,
+		MinTimeout: 3 * time.Second,
+	}
+
+	_, err = stateConf.WaitForState()
+	if err != nil {
+		return fmt.Errorf("Error creating HuaweiCloud CCEAddon: %s", err)
+	}
+	d.SetId(create.Metadata.Id)
```

Comment:
do SetId before waiting the state, as the resource had been created.

Reply:
fixed